src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrBackflower.cs

```diff
@@ -0,0 +1,323 @@
+    private readonly ILocalGitClient _localGitClient;
+    private readonly IVersionDetailsParser _versionDetailsParser;
+    private readonly IVmrPatchHandler _vmrPatchHandler;
+    private readonly IProcessManager _processManager;
+    private readonly IWorkBranchFactory _workBranchFactory;
+    private readonly IFileSystem _fileSystem;
+    private readonly ILogger<VmrCodeflower> _logger;
+
+    public VmrBackflower(
+        IVmrInfo vmrInfo,
+        ISourceManifest sourceManifest,
+        IVmrDependencyTracker dependencyTracker,
+        IDependencyFileManager dependencyFileManager,
+        ILocalGitClient localGitClient,
+        IVersionDetailsParser versionDetailsParser,
+        IVmrPatchHandler vmrPatchHandler,
+        IProcessManager processManager,
+        IWorkBranchFactory workBranchFactory,
+        IFileSystem fileSystem,
+        ILogger<VmrCodeflower> logger)
+        : base(vmrInfo, sourceManifest, dependencyTracker, localGitClient, versionDetailsParser, processManager, fileSystem, logger)
+    {
+        _vmrInfo = vmrInfo;
+        _sourceManifest = sourceManifest;
+        _dependencyFileManager = dependencyFileManager;
+        _localGitClient = localGitClient;
+        _versionDetailsParser = versionDetailsParser;
+        _vmrPatchHandler = vmrPatchHandler;
+        _processManager = processManager;
+        _workBranchFactory = workBranchFactory;
+        _fileSystem = fileSystem;
+        _logger = logger;
+    }
+
+    public async Task<string?> FlowBackAsync(
+        string mapping,
+        NativePath targetRepo,
+        string? shaToFlow = null,
+        bool discardPatches = false,
+        CancellationToken cancellationToken = default)
+    {
+        if (shaToFlow is null)
+        {
+            shaToFlow = await _localGitClient.GetShaForRefAsync(_vmrInfo.VmrPath, Constants.HEAD);
+        }
+        else
+        {
+            await CheckOutVmr(shaToFlow);
+        }
+
+        var branchName = await FlowCodeAsync(
+            isBackflow: true,
+            targetRepo,
+            mapping,
+            shaToFlow,
+            discardPatches,
+            cancellationToken);
+
+        if (branchName is null)
+        {
+            // TODO: We should still probably update package versions or at least try?
+            // Should we clean up the repos?
+            return null;
+        }
+
+        await UpdateVersionDetailsXml(targetRepo, shaToFlow, cancellationToken);
+        return branchName;
+    }
+
+    protected override async Task<string?> SameDirectionFlowAsync(
+        SourceMapping mapping,
+        string shaToFlow,
+        NativePath repoPath,
+        Codeflow lastFlow,
+        bool discardPatches,
+        CancellationToken cancellationToken)
+    {
+        var isBackflow = lastFlow is Backflow;
+        var shortShas = $"{Commit.GetShortSha(lastFlow.SourceSha)}-{Commit.GetShortSha(shaToFlow)}";
+        var branchName = $"codeflow/{lastFlow.GetType().Name.ToLower()}/{shortShas}";
```

Comment:
Strong recommendation to avoid using reflection unless necessary (for many reasons). Introduce a `string Type` property on the CodeFlow and its  descendants and use it instead.